command/agent/consul/syncer.go

```diff
@@ -355,7 +376,12 @@ func (c *Syncer) Shutdown() error {
 	}
 
 	// De-register all the services from Consul
-	for serviceID := range c.trackedServices {
+	services, err := c.queryAgentServices()
```

Comment:
I would change the comment above:

`// De-register all services registered by this syncer from Consul`